dtfs-central-api/src/v1/controllers/utilisation-report-service/get-fee-record-details-by-id.controller/index.test.ts

```diff
@@ -0,0 +1,103 @@
+import httpMocks from 'node-mocks-http';
+import { HttpStatusCode } from 'axios';
+import { when } from 'jest-when';
+import { FeeRecordEntityMockBuilder, UtilisationReportEntityMockBuilder } from '@ukef/dtfs2-common';
+import { getFeeRecordDetailsById, GetFeeRecordDetailsResponseBody } from '.';
+import { mapToFeeRecordDetails } from './helpers';
+import { FeeRecordRepo } from '../../../../repositories/fee-record-repo';
+
+console.error = jest.fn();
+
+jest.mock('./helpers');
+
+describe('get-fee-record-details-by-id.controller', () => {
+  describe('getFeeRecordDetailsById', () => {
+    const reportId = 3;
+    const feeRecordId = 14;
+
+    const getHttpMocks = () =>
+      httpMocks.createMocks({
+        params: { reportId: reportId.toString(), feeRecordId: feeRecordId.toString() },
+      });
+
+    const findOneSpy = jest.spyOn(FeeRecordRepo, 'findOne');
```

Comment:
Please can we mock 'findOneByIdAndReportIdWithReport' instead of 'findOne', the details of how that is translated into a typeorm query do not lie in this controller and testing it here adds unnecessary complexity to these tests.